src/renderer/components/RuntimeAdapter.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useEffect } from 'react'
+
+import useMessage from '../hooks/useMessage'
+import useProcessQr from '../hooks/useProcessQr'
+import { clearNotificationsForChat } from '../system-integration/notifications'
+import { runtime } from '../runtime'
+import { selectedAccountId } from '../ScreenController'
+
+import type { PropsWithChildren } from 'react'
+
+/**
+ * Helper component to hook React methods into the external "runtime". This
+ * allows us to interact with the underlying Electron runtime and
+ * operating system.
+ */
+export default function RuntimeAdapter({ children }: PropsWithChildren<{}>) {
```

Comment:
I came up with this `RuntimeAdapter` as a glue between the "external runtime world" and React. Otherwise there's no way we get access to React hooks etc from the outside.

So far this has been solved with using callbacks on `window` which then somewhat call React code but all of this will then happen outside of the React lifetime and state cycle. Doing it like this instead could be a nice pattern for future refactorings.

Note that this is only useful for calling React code whenever an external event took place. For the other way around (calling external functions when an React event took place) we should be already set.